src/imageCache.js

```diff
@@ -170,46 +213,34 @@ export function processExternalGraphicSymbolizers(
   for (let k = 0; k < rules.length; k += 1) {
     const rule = rules[k];
 
-    let symbolizer;
-    let exgraphic;
+    const allSymbolizers = getAllSymbolizers(rule);
+    allSymbolizers.forEach(symbolizer => {
+      externalGraphicPaths.forEach(path => {
+        const exgraphic = getByPath(symbolizer, path);
+        if (!exgraphic) {
+          return;
+        }
 
-    if (
-      rule.pointsymbolizer &&
-      rule.pointsymbolizer.graphic &&
-      rule.pointsymbolizer.graphic.externalgraphic
-    ) {
-      symbolizer = rule.pointsymbolizer;
-      exgraphic = rule.pointsymbolizer.graphic.externalgraphic;
-    } else if (
-      rule.polygonsymbolizer &&
-      rule.polygonsymbolizer.fill &&
-      rule.polygonsymbolizer.fill.graphicfill &&
-      rule.polygonsymbolizer.fill.graphicfill.graphic &&
-      rule.polygonsymbolizer.fill.graphicfill.graphic.externalgraphic
-    ) {
-      symbolizer = rule.polygonsymbolizer;
-      exgraphic =
-        rule.polygonsymbolizer.fill.graphicfill.graphic.externalgraphic;
-    } else {
-      continue;
-    }
-
-    const imageUrl = exgraphic.onlineresource;
-    if (!(imageUrl in imageLoadState)) {
-      // Start loading the image and set image load state on the symbolizer.
-      imageLoadState[imageUrl] = IMAGE_LOADING;
-      loadExternalGraphic(
-        imageUrl,
-        imageLoadState,
-        featureTypeStyle,
-        imageLoadedCallback
-      );
-    } else if (
-      // Change image load state on the symbolizer if it has changed in the meantime.
-      symbolizer.__loadingState !== imageLoadState[imageUrl]
-    ) {
-      updateExternalGraphicRule(rule, imageUrl, imageLoadState[imageUrl]);
-    }
+        // When an external graphic has been found inside a symbolizer,
+        // either start loading the image, or check if the load state of the image has changed.
+        const imageUrl = exgraphic.onlineresource;
+        if (!(imageUrl in imageLoadState)) {
+          // Start loading the image and set image load state on the symbolizer.
+          imageLoadState[imageUrl] = IMAGE_LOADING;
+          loadExternalGraphic(
```

Comment:
It seems you choose to set the imageloadstate inside the sld symbolizer object. Can this concern be moved into a separate object?

Reply:
I will move checking image load states to getters/setters, just like for the cached image, which also has getter/setter for a global cache. I think that will improve the clarity of what's happening here. After doing that, I think I can stop storing the load state flag inside the symbolizer.